File: versions/forge1710/src/main/scala/com/idkidknow/mcreallink/forge1710/mixin/complement/BroadcastingMessage.scala
package com.idkidknow.mcreallink.forge1710.mixin.complement

import net.minecraft.util.IChatComponent

object BroadcastingMessage {
  private val _ignoreMessage = ThreadLocal.withInitial(() => false)
  def ignoreMessage: Boolean = _ignoreMessage.get()
  @SuppressWarnings(Array("org.wartremover.warts.Var"))
  var callback: IChatComponent => Unit = { _ => () }
  def getCallback: java.util.function.Consumer[IChatComponent] = { c => callback(c) }

  def ignoreTemporarily(action: () => Unit): Unit = {
    _ignoreMessage.set(true)
    action()
    _ignoreMessage.set(false)
  }
}

File: versions/forge1710/src/common/java/com/idkidknow/mcreallink/forge1710/mixin/complement/BroadcastingMessage.java
package com.idkidknow.mcreallink.forge1710.mixin.complement;

import net.minecraft.util.IChatComponent;

import java.util.function.Consumer;

public class BroadcastingMessage {
    private static final ThreadLocal<Boolean> ignoreMessage = ThreadLocal.withInitial(() -> false);
    public static boolean isIgnoreMessage() {
        return ignoreMessage.get();
    }
    private static Consumer<IChatComponent> callback = component -> {};
    public static Consumer<IChatComponent> getCallback() {
        return callback;
    }
    public static void setCallback(Consumer<IChatComponent> callback) {
        BroadcastingMessage.callback = callback;
    }

    public static void ignoreTemporarily(Runnable action) {
        ignoreMessage.set(true);
        action.run();
        ignoreMessage.set(false);
    }
}
